### PR TITLE
hw3: Unit Testing

### PR DESCRIPTION
**Pull request by terryyLi**
Hi Jeremy,

 I've pushed hw3 readme. Can you take a look at it? Please let me know if there's any questions. Thank you.

Terry

### CODE REVIEW

File: assignments/hw3.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- `ui.ts`: you will not need to test UI.
+
+Please note that the last four exclusions would not be industry-standard. They are nontrivial to test, though. We will show you later in the course how you can test such functionality.
+
+Our reference implementation has about 40 tests.  As a hint, for `cardorganizer.ts`, there are many different possible combinations of sorters and repeaters.  You should _not_ need to test all combinations to achieve suitable specification coverage!
+
+
+### Part 3: Structural testing (Java)
+
+For the Java implementation, write structural tests for the same files you tested in Part 2 to achieve 100% *branch* coverage. You do not need to test files that were excluded in Part 2 and branch coverage is not required for files that are not part of your testing scope. We will not evaluate your test quality for the source code with injected bugs; the goal is to maximize code coverage for the relevant files.
```
Comment (jlacomis) [inline assignments/hw3.md:58]: You can also exclude code in the `Achievements` directory.

File: assignments/hw3.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+In section **Testing strategy** briefly describe whether you followed any specific test case design strategy, such as boundary value analysis, when creating tests. Briefly explain, why you did or did not use these techniques. (about 1 paragraph)
+
+In section **Specification vs structure testing** briefly reflect on your experience with the two different testing approaches. Was one harder or better than the other for you? (about 1 or 2 paragraphs)
+
+If you used AI tools such as ChatGPT or Copilot for any part of homework 1 or 2, add a section **AI Tools** describing what you tried them for and whether you found them useful (open ended).
+
+## Submitting your work
+
+Always submit all code to GitHub. Once you have pushed your final code there, submit a link to your final commit in the format `https://github.com/CMU-17-214/<reponame>/commit/<commitid>`  on Canvas (the link identifies the right commit and branch for us, in case you do not work in the main branch). You can get this link easily when you click on the last commit (above the list of files) in the GitHub web interface.
```
Comment (jlacomis) [inline assignments/hw3.md:78]: Their code is in the `CMU-17-214-Students` organization

**Review by jlacomis — CHANGES_REQUESTED** [review]
These changes and I can accept the PR

File: assignments/hw3.md (hunk 1 above)
Comment (jlacomis) [reply assignments/hw3.md]: Fixed

File: assignments/hw3.md (hunk 2 above)
Comment (jlacomis) [reply assignments/hw3.md]: Resolved

**Review by jlacomis — APPROVED** [review]
Looks good now
